resource-managers/mesos/src/test/scala/org/apache/spark/scheduler/cluster/mesos/MesosFineGrainedSchedulerBackendSuite.scala

```diff
@@ -79,6 +79,43 @@ class MesosFineGrainedSchedulerBackendSuite
     backend.start()
   }
 
+  test("mesos supports checkpointing") {
+    val conf = new SparkConf
+    conf.set("spark.mesos.checkpoint", "true")
+    conf.set("spark.mesos.failoverTimeout", "10")
+    conf.set("spark.mesos.driver.webui.url", "http://webui")
+
+    val sc = mock[SparkContext]
+    when(sc.conf).thenReturn(conf)
+    when(sc.sparkUser).thenReturn("sparkUser1")
+    when(sc.appName).thenReturn("appName1")
+
+    val taskScheduler = mock[TaskSchedulerImpl]
+    val driver = mock[SchedulerDriver]
+    when(driver.start()).thenReturn(Protos.Status.DRIVER_RUNNING)
+
+    val backend = new MesosFineGrainedSchedulerBackend(taskScheduler, sc, "master") {
+      override protected def createSchedulerDriver(
+        masterUrl: String,
+        scheduler: Scheduler,
+        sparkUser: String,
+        appName: String,
+        conf: SparkConf,
+        webuiUrl: Option[String] = None,
+        checkpoint: Option[Boolean] = None,
+        failoverTimeout: Option[Double] = None,
+        frameworkId: Option[String] = None): SchedulerDriver = {
+        markRegistered()
+        assert(checkpoint.contains(true))
+        assert(failoverTimeout.contains(10))
```

Comment:
I know this does not cause the test to fail but probably we should use `10.0` here as well, just for conformity. Incidentally, `Option.contains` uses `==` instead of `equals` for comparison, that's why it passes; I did not know that 😅